teku/src/main/java/tech/pegasys/teku/cli/options/BeaconNodeDataOptions.java

```diff
@@ -107,6 +107,9 @@ public void configure(final TekuConfiguration.Builder builder) {
   }
 
   private DatabaseVersion parseDatabaseVersion() {
+    if (dataStorageFrequency == 1) {
+      return DatabaseVersion.LEVELDB_TREE;
+    }
     return DatabaseVersion.fromString(createDbVersion).orElse(DatabaseVersion.DEFAULT_VERSION);
```

Comment:
We don't want to force switching to LEVELDB_TREE if something else is explicitly specified - the user may be trying to force using rocksdb or something like that (or it may be us trying to test). Given it's a hidden option I'd probably just have the field default to null and then you can select leveldb2 or leveldb-tree based on the frequency when version is `null` and use whatever is specified when it isn't. It won't show the default value then but I don't think that's a big deal for a hidden option.